src/main/java/uk/nhs/hee/tis/revalidation/connection/mapper/ConnectionLogMapper.java

```diff
@@ -0,0 +1,15 @@
+package uk.nhs.hee.tis.revalidation.connection.mapper;
+
+import java.util.UUID;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import uk.nhs.hee.tis.revalidation.connection.dto.ConnectionLogDto;
+import uk.nhs.hee.tis.revalidation.connection.entity.ConnectionLog;
+
+@Mapper(componentModel = "spring", imports = UUID.class)
+public interface ConnectionLogMapper {
+  @Mapping(target = "id", expression = "java(UUID.randomUUID().toString())")
```

Comment:
I feel it a bit strange to generate the UUID for id in the mapper. 
I remember in TCS, we usually use a id generator in the entity class. But this id is a string, I'm not too sure if it would be easy to do the same thing.